libs/core/langchain_core/messages/base.py

```diff
@@ -95,9 +95,13 @@ def merge_content(
         else:
             return_list: List[Union[str, Dict]] = [first_content]
             return return_list + second_content
-    # If both are lists, merge them naively
     elif isinstance(second_content, List):
-        return first_content + second_content
+        # If both are lists
+        merged_list = merge_lists(first_content, second_content)
```

Comment:
only break I can think of is if one list has a dict item with key `"index"` and the other doesn't. e.g., `merge_lists([{"a": 3}], [{"a": 3, "index": 2}])` will raise, whereas before we would just add the lists. but unclear to me if this condition happens.

Reply:
Yeah unclear what would make sense - I think ok to raise? I can add a test case

Reply:
Ah it's the opposite right now. I think implicitly appending is ok too and actually may make more sense

Reply:
Unresolve if you disagree